test/tests/performance/pbs_qsub_performance.py

```diff
@@ -37,47 +37,58 @@
 # "OpenPBS®", "PBS Professional®", and "PBS Pro™" and Altair's logos is
 # subject to Altair's trademark licensing policies.
 
-
+import subprocess
 from tests.performance import *
 
 
 class TestQsubPerformance(TestPerformance):
-    """
-    This test suite contains tests of qsub performance
-    """
 
     def setUp(self):
         TestPerformance.setUp(self)
         attr = {'scheduling': 'False'}
         self.server.manager(MGR_CMD_SET, SERVER, attr)
 
-    @timeout(400)
-    def test_submit_large_env(self):
+    def submit_jobs(self, qsub_exec_arg=None, env=None):
         """
-        submission of 1000 jobs
-        before and after exporting variable
-        to check submission performance
+        Submits n num of jobs according to the arguments provided
+        and returns submission time
+        :param qsub_exec_arg: Arguments to qsub.
+        :type qsub_exec_arg: String. Defaults to None.
+        :param env: Environment variable to be set before submittign job.
+        :type env: Dictionary. Defaults to None.
         """
+        user = self.du.get_current_user()
+        qsub_path = os.path.join(
+                  self.server.pbs_conf['PBS_EXEC'], 'bin', 'qsub')
+        common_cmd = ' -u ' + user + ' ' + qsub_path
+        if qsub_exec_arg is not None:
+            job_sub_arg = "sudo -E" + common_cmd + ' ' + qsub_exec_arg
+        else:
+            job_sub_arg = 'sudo' + common_cmd
 
-        start_time1 = time.time()
-        for _ in range(1000):
-            j = Job(TEST_USER)
-            self.server.submit(j)
-        end_time1 = time.time()
-
-        os.environ['VARIABLE'] = 'b' * 130000
+        job_sub_arg += ' -- /bin/sleep 100'
 
-        start_time2 = time.time()
+        start_time = time.time()
         for _ in range(1000):
-            j1 = Job(TEST_USER)
-            self.server.submit(j1)
-        end_time2 = time.time()
+            subprocess.call(job_sub_arg, shell=True, env=env)
```

Comment:
Can't you do this through run_cmd()?  I thought it was extended a while back to take an environment.